src/build_rock/configure/generate_build_matrix.py

```diff
@@ -0,0 +1,139 @@
+from ...shared import github_output
+
+
+class MATRIX_NAMES(Enum):
+    RUNNER = "runner-build-matrix"
+    LPCI = "lpci-build-matrix"
+
+
+class MissingArchSupport(Exception):
+    pass
+
+
+def get_target_archs(rockcraft: dict) -> list:
+    """get list of target architectures from rockcraft project definition"""
+
+    rock_platforms = rockcraft["platforms"]
+
+    target_archs = set()
+
+    for platf, values in rock_platforms.items():
+
+        if isinstance(values, dict) and "build-for" in values:
+            # TODO: Should this be "build-on"?
+            # https://documentation.ubuntu.com/rockcraft/en/latest/reference/rockcraft.yaml/#platforms-entry-build-on
+            if isinstance(arches := values["build-for"], list):
+                target_archs.update(arches)
+            elif isinstance(values, str):
+                target_archs.add(arches)
+        else:
+            target_archs.add(platf)
+
+    return target_archs
+
+
+def configure_matrices(target_archs: list, arch_map: dict, lp_fallback: bool) -> dict:
+    """Sort build into appropriate build matrices"""
+
+    # map configuration to individual job matrices
+    build_matrices = {name.value: {"include": []} for name in MATRIX_NAMES}
+
+    # Check if we have runners for all supported architectures
+    if missing_archs := set(target_archs) - set(arch_map.keys()):
+
+        # raise exception if we cannot fallback to LP builds
+        if not lp_fallback:
+            raise MissingArchSupport(
+                f"Missing support for runner arches: {missing_archs}"
+            )
+
+        # configure LP build
+        build_matrices[MATRIX_NAMES.LPCI.value]["include"].append(
+            {"architecture": "-".join(set(target_archs))}
+        )
+
+    else:
+        # configure runner matrix for list of supported runners
+        for runner_arch, runner_name in arch_map.items():
+            if runner_arch in target_archs:
+                build_matrices[MATRIX_NAMES.RUNNER.value]["include"].append(
+                    {"architecture": runner_arch, "runner": runner_name}
+                )
+
+    return build_matrices
+
+
+def set_build_config_outputs(
+    rock_name: str, build_matrices: dict, output_path: Optional[str] = None
+):
+    """Update GITHUB_OUTPUT with build configuration."""
+
+    outputs = {"rock-name": rock_name, **build_matrices}
+
+    # set default when not testing
+    if output_path is None:
+        output_path = os.environ["GITHUB_OUTPUT"]
+
+    with open(output_path, "a") as fh:
+        github_output.write(fh, **outputs)
+
+
+def main():
+    parser = argparse.ArgumentParser()
+
+    parser.add_argument(
+        "--rockfile-directory",
+        help="Path where to directory containing rockcraft.yaml.",
+        required=True,
+    )
+
+    parser.add_argument(
+        "--lpci-fallback",
+        help="Revert to lpci if architectures are not supported. <true|false>",
+        required=True,
+        type=str,
+    )
```

Comment:
Done. Thanks!

Reply:
I like the idea of breaking it out into a function to cast the type. Maybe it would be easier to match the string directly, that way I can still have somewhat flexible inputs.